### DISCUSSION TITLE
Autocomplete with TValue as Guid fails validation

### DISCUSSION THREAD

**Discussion by mdmontesinos** [Megabit/Blazorise#5133] [Q&A]
Using Autocomplete component with SelectionMode to Multiple and a TValue of type Guid does not correctly apply validation.

When replacing the Validator and printing to console the value, you always get the following:

![image](https://github.com/Megabit/Blazorise/assets/90258222/27f2ca4b-bfd4-40e6-bebc-866891bb9971)

It doesn't matter how many options you've chosen, the internal value used for validation is always a default Guid converted to string.

Therefore, it's not possible to validate whether the value is empty or not with the ValidationRule.IsNotEmpty.

After taking a brief look at the implementation, I believe the Guid (which does not accept null values) is always converted to its default value and ToString.

**Comment by David-Moreira** [comment]
Converting to issue, possible bug

**Reply by David-Moreira** [reply]
#5135

**Reply by David-Moreira** [reply]
Hello @mdmontesinos 

What is so wrong about this? It seems to me the behaviour might be correct. If you have chosen a `Guid` as the `TValue`, it's default value will be in fact `Guid.Empty`.
If you want an actual null, have you tried binding a nullable `Guid`?

As for the `ValidationRule.IsNotEmpty` it is stated that this validation helper checks a string.
> Check if the string is not null or empty.
![image](https://github.com/Megabit/Blazorise/assets/22283161/e4b2d57d-288a-41a7-b7ee-6374a287ef5d)

You may go ahead and do your own validation helper that checks for an Empty `Guid` instead. 
Something along these lines should work:

`public static void IsGuidNotEmpty( ValidatorEventArgs e ) =>   e.Status = (e.Value is Guid valueAsGuid && !valueAsGuid.Equals(Guid.Empty)) ? ValidationStatus.Success : ValidationStatus.Error;
`

**Reply by mdmontesinos** [reply]
You're right about the ValidationRule.IsNotEmpty not making sense as the Guid is always converted to its empty value.

However, when implementing a custom validator, the Value received in the ValidatorEventArgs is always the Guid.Empty.

I've binded the SelectedTextsChanged and SelectedValuesChanged events, and you can see that the Validator value is always empty:

![image](https://github.com/Megabit/Blazorise/assets/90258222/b4589039-3a9d-48bb-94f7-be0605c0457c)

Perhaps this is not related with the Guid itself, but with the SelectionMode multiple. I've tested changing the TValue to a string (and casted the Guid to string) and I still get the same result, no validator value:

![image](https://github.com/Megabit/Blazorise/assets/90258222/da588104-0077-49b7-9b87-92099f729921)

Isn't it possible to validate a multiple type autocomplete?
